### PR TITLE
PP 394 add return assistant screen events

### PR DESCRIPTION
**Pull request by ValentinaIancu-Gini**
Add return assistant screen events in

1. Onboarding return assistant screen
2. Return assistant  screen
3. Edit article screen
4. Help screen

### CODE REVIEW

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsScreen.swift [hunk 1]
```diff
@@ -19,4 +19,7 @@ enum AnalyticsScreen: String {
     case onboardingMultipage = "onboarding_multiple_pages"
     case onboardingQRcode = "onboarding_qr_code"
     case onboardingCustom = "onboarding_custom_" // e.g: onboarding_custom_1, onboarding_custom_2
+    case onboardingReturnAssitant = "onboarding_return_assistant"
+    case returnAssitant = "return_assistant"
```
Comment (zladzeyka) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsScreen.swift:23]:
```suggestion
    case returnAssistant = "return_assistant"
```

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsScreen.swift [hunk 2]
```diff
@@ -19,4 +19,7 @@ enum AnalyticsScreen: String {
...
+    case editReturnAssitant = "edit_return_assistant"
```
Comment (zladzeyka) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsScreen.swift:24]:
```suggestion
    case editReturnAssistant = "edit_return_assistant"
```

File: BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/ReturnAssistant/EditLineItem/EditLineItemViewModel.swift [hunk 3]
```diff
@@ -16,6 +15,7 @@ protocol EditLineItemViewModelDelegate: AnyObject {
 final class EditLineItemViewModel {
     weak var delegate: EditLineItemViewModelDelegate?
     private var lineItem: DigitalInvoice.LineItem
+    var itemsChanged: [LineItemAnalytics] = []
```
Comment (zladzeyka) [inline BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/ReturnAssistant/EditLineItem/EditLineItemViewModel.swift:18]: [Clarification]: can we make this var private?

File: BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/ReturnAssistant/DigitalInvoice/DigitalInvoiceCoordinator.swift [hunk 4]
```diff
@@ -21,6 +20,7 @@ final class DigitalInvoiceCoordinator: Coordinator {
     private var digitalInvoiceViewController: DigitalInvoiceViewController?
     private var digitalInvoiceViewModel: DigitalInvoiceViewModel?
 
+    // TODO: do we still need this?!
```
[the remark this answers is not in the document]
Comment (ValentinaIancu-Gini) [reply BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/ReturnAssistant/DigitalInvoice/DigitalInvoiceCoordinator.swift:23]: I will do it when the parent branch will be merged in the "feature" branch. thank you!

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsProperties.swift [hunk 5]
```diff
@@ -56,4 +61,8 @@ enum AnalyticsPropertyKey: String {
     case helpItems = "help_items"
     case itemTapped = "item_tapped"
     case customOnboardingTitle = "custom_onboarding_title"
+    case documentId = "document-id"
```
[the remark this answers is not in the document]
Comment (ValentinaIancu-Gini) [reply CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsProperties.swift:64]: It should be `_` thanks for catching this!

File: BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/ReturnAssistant/DigitalInvoice/DigitalInvoiceViewController.swift [hunk 6]
```diff
@@ -221,7 +220,18 @@ final class DigitalInvoiceViewController: UIViewController {
         viewModel.shouldShowOnboarding()
     }
 
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        if presentedViewController == nil {
+            // Send a 'screenShown' event when returning from the help screen or any other screen
+            // that was pushed over this view controller.
```
[the remark this answers is not in the document]
Comment (ValentinaIancu-Gini) [reply BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/ReturnAssistant/DigitalInvoice/DigitalInvoiceViewController.swift:227]: I've improved the comment. 
Help screen is pushed(the action it self) when tapping the help button from the Return Assistant screen. That's what I meant by that.

File: BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/ReturnAssistant/DigitalInvoice/DigitalInvoiceCoordinator.swift [hunk 7]
```diff
@@ -98,16 +100,26 @@ extension DigitalInvoiceCoordinator: EditLineItemViewModelDelegate {
...
+        navigationController.dismiss(animated: true) {
+            self.digitalInvoiceViewController?.sendAnalyticsScreenShown()
+        }
     }
 
     func didCancel(on viewModel: EditLineItemViewModel) {
-        navigationController.dismiss(animated: true)
+        AnalyticsManager.track(event: .closeTapped, screenName: .editReturnAssitant)
+        navigationController.dismiss(animated: true) {
+            self.digitalInvoiceViewController?.sendAnalyticsScreenShown()
```
[the remark this answers is not in the document]
Comment (ValentinaIancu-Gini) [reply BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/ReturnAssistant/DigitalInvoice/DigitalInvoiceCoordinator.swift:122]: Sure. It's safer like that.

File: BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/ReturnAssistant/EditLineItem/EditLineItemViewModel.swift (hunk 3 above)
Comment (ValentinaIancu-Gini) [reply BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/ReturnAssistant/EditLineItem/EditLineItemViewModel.swift:18]: I've made just the `set` private. I'm using it outside of this file.

File: BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/ReturnAssistant/DigitalInvoice/DigitalInvoiceCoordinator.swift [hunk 8]
```diff
@@ -98,16 +100,26 @@ extension DigitalInvoiceCoordinator: EditLineItemViewModelDelegate {
     func didSave(lineItem: DigitalInvoice.LineItem, on viewModel: EditLineItemViewModel) {
         guard let invoice = digitalInvoiceViewModel?.invoice else { return }
 
+        var eventProperties: [AnalyticsProperty] = []
```
[the remark this answers is not in the document]
Comment (ValentinaIancu-Gini) [reply BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/ReturnAssistant/DigitalInvoice/DigitalInvoiceCoordinator.swift:103]: The array is changed inside the if:
````
....
                eventProperties.append(AnalyticsProperty(key: .itemsChanged,
                                                         value: viewModel.itemsChanged.map { return $0.rawValue }))

.....
 ```

File: BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/ReturnAssistant/Onboarding/DigitalInvoiceOnboardingViewController.swift [hunk 9]
```diff
@@ -1,32 +1,33 @@
...
+    @IBOutlet private weak var contentView: UIView!
+    @IBOutlet private weak var topImageView: OnboardingImageView!
+    @IBOutlet private weak var firstLabel: UILabel!
+    @IBOutlet private weak var secondLabel: UILabel!
+    @IBOutlet private weak var scrollView: UIScrollView!
+    @IBOutlet private weak var doneButton: MultilineTitleButton!
+    @IBOutlet private weak var scrollViewTopConstraint: NSLayoutConstraint!
+    @IBOutlet private weak var scrollViewBottomAnchor: NSLayoutConstraint!
+
+    weak var delegate: DigitalInvoiceOnboardingViewControllerDelegate!
```
Comment (mrkulik) [inline BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/ReturnAssistant/Onboarding/DigitalInvoiceOnboardingViewController.swift:24]: [Suggestion] @ValentinaIancu-Gini in dismissViewController() func delegate is marked as optional, and here its force-unwrapped. What do you think about marking it here also as optional. or remove optional inside func?

**Review by mrkulik — CHANGES_REQUESTED** [review]
@ValentinaIancu-Gini just 1 quick suggestion from my side. Great job🙌

File: BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/ReturnAssistant/DigitalInvoice/DigitalInvoiceCoordinator.swift [hunk 10]
```diff
@@ -98,16 +100,26 @@ extension DigitalInvoiceCoordinator: EditLineItemViewModelDelegate {
...
         if invoice.lineItems.indices.contains(viewModel.index) {
```
[the remark this answers is not in the document]
Comment (ValentinaIancu-Gini) [reply BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/ReturnAssistant/DigitalInvoice/DigitalInvoiceCoordinator.swift:104]: I will do it. Thank you for your suggestion!

File: BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/ReturnAssistant/Onboarding/DigitalInvoiceOnboardingViewController.swift (hunk 9 above)
Comment (ValentinaIancu-Gini) [reply BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/ReturnAssistant/Onboarding/DigitalInvoiceOnboardingViewController.swift:24]: Sure I will mark it optional. Thanks for the catch!

**Comment by ValentinaIancu-Gini** [comment]
@zladzeyka and @mrkulik 
Thank you for your great input!
I did some updates. You can check now once again the pr. 
thank you!

**Review by mrkulik — APPROVED** [review]
@ValentinaIancu-Gini great job!

**Review by zladzeyka — APPROVED** [review]
Great job!

**Comment by ValentinaIancu-Gini** [comment]
> Great job!

Thanks @zladzeyka !

**Pull request merged**